### PR TITLE
[hpal] add Daybreak talent

### PR DESCRIPTION
**Pull request by mfield**
### Description
Adds Daybreak talent. Removes some misattributed healing from  Glistening Radiance.
Maybe conflicts with #6204 i can resolve once one is merged.

### Testing

live
image

this pr
image

### CODE REVIEW

**Comment by abelito75** [comment]
I think my glimmer re-write kind of invalidates most of this
There are things that I don't fully cover such as mana but everything else should be covered in my other PR

**Comment by emallson** [comment]
might need to PR yours @abelito75 and then compare notes?

**Comment by abelito75** [comment]
Its this guy! #6206

It tracks more glimmer proc sources and has a great UI
It does miss out on the MP5 and CastLinkNormalizer but I think outside of that it does everything and fixes some bugs
We talked about it a bit in discord but I don't think we came to a decision

**Comment by mfield** [comment]
let's figure out what we want in #6206 first. i left some notes there. but i'm not attached if this gets dropped nbd.

**Comment by mfield** [comment]
taking this opportunity to break down per cast since most of the overview is in the Glimmer box
image

File: src/analysis/retail/paladin/holy/modules/talents/Daybreak.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      Events.cast.spell(TALENTS.DAYBREAK_TALENT).by(SELECTED_PLAYER),
+      this.onCast,
+    );
+  }
+
+  onCast(event: CastEvent) {
+    this.castEvents.push(event);
+
+    const manaEvents = GetRelatedEvents(event, DAYBREAK_MANA);
+    manaEvents.forEach((e: AnyEvent) => {
```
Comment (abelito75) [inline src/analysis/retail/paladin/holy/modules/talents/Daybreak.tsx]: IDK if its possible but is there any way we can not do `AnyEvent`... maybe `.filter()` before hand?
Maybe an ugly cast on the whole array?

File: src/analysis/retail/paladin/holy/modules/talents/Daybreak.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                <th>Overhealing</th>
+              </tr>
+            </thead>
+            <tbody>
+              {Object.values(this.castEvents).map((cast, i) => (
+                <tr key={i}>
+                  <td>{this.owner.formatTimestamp(cast.timestamp)}</td>
+                  <td>{GetRelatedEvents(cast, DAYBREAK_MANA).length}</td>
+                  <td>
+                    {formatNumber(
```
Comment (abelito75) [inline src/analysis/retail/paladin/holy/modules/talents/Daybreak.tsx]: should probably do what you did for mana here and break it into functions. A lot easier to read IMO

**Review by abelito75 — CHANGES_REQUESTED** [review]

File: src/analysis/retail/paladin/holy/modules/talents/Daybreak.tsx (hunk 1 above)
Comment (mfield) [reply src/analysis/retail/paladin/holy/modules/talents/Daybreak.tsx]: harder to change the type here, but easy to change the next line to `if (e.type === EventTypes.ResourceChange)` which i think is nicer than the explicit cast. decent resolution?

Comment (abelito75) [reply src/analysis/retail/paladin/holy/modules/talents/Daybreak.tsx]: That works but also not the end of the world if you just wanna send it

**Review by abelito75 — APPROVED** [review]

**Pull request merged**